tdms/tests/system/utils.py

```diff
@@ -36,7 +36,20 @@ def __init__(self, filepath: Union[str, Path]):
         super().__init__()
 
         with h5py.File(filepath, "r") as file:
-            self.update({k: self.to_numpy_array(v) for k, v in file.items()})
+            self.update({k: self.to_numpy_array(v) for k, v in self.traverse(file)})
+
+    def traverse(self, file_or_group, prefix=""):
```

Comment:
Add typing to this new method?

Reply:
I didn't because it was going to be something epic like...
```py
def traverse(self, file_or_group: Union[h5py.File, h5py.Group], prefix: str = "") -> Generator[str, h5py.Dataset]:
```
I'm perfectly happy to do this. It's just... a lot.

Reply:
I think that's not too bad, and would be +1 to adding it.